### ISSUE TITLE
Make adjustment easier - Input box contains values of properties

### ISSUE THREAD

**Issue by jukrb0x** [lwouis/alt-tab-macos#864]
**Is your feature suggestion related to a problem? Please describe.**
When adjusting values for appearance, it's hard to control the value bar for a precise adjustment.

![image](https://user-images.githubusercontent.com/15688641/110434417-c1faac00-80ec-11eb-8efb-ef663b17d62d.png)

**Describe the solution you'd like**
The right side indicates the value of properties, which can be made in an input box that contains particular values.

**Describe alternatives you've considered**
N/A

**Comment by jukrb0x** [comment]
Thanks @lwouis for responding!

User can type in a wrong way such as letters or negative values, but the text box can do a test with input values right? Say if user input is not in the range of valid data then it will not be accepted by the text box.

I totally agree the visually sliders would help to understand a lot, what I mean is, we can keep the setting slider, but the values at the right hand side of the slider, can be made in a text box contains the value of property controlled by the slider. That is, user can type a precise value (validity should be tested when typing) and also can slide the setting bar for choosing a proper position. 

Also, I've tried the arrow keys to handle data and it doesn't work for me, can u elaborate it.

**Comment by jukrb0x** [comment]
> Thanks @lwouis for responding!
> 
> User can type in a wrong way such as letters or negative values, but the text box can do a test with input values right? Say if user input is not in the range of valid data then it will not be accepted by the text box.
> 
> I totally agree the visually sliders would help to understand a lot, what I mean is, we can keep the setting slider, but the values at the right hand side of the slider, can be made in a text box contains the value of property controlled by the slider. That is, user can type a precise value (validity should be tested when typing) and also can slide the setting bar for choosing a proper position.
> 
> Also, I've tried the arrow keys to handle data and it doesn't work for me, I clicked the circle of slider and press the arrow keys. can u please elaborate how to do it.

For instance, I took a design software as an example, the property Width can be adjusted as the slider bar changes or the value in the text box changes, these two indicate the same.

![image](https://user-images.githubusercontent.com/15688641/110497854-ca290a80-8131-11eb-9faf-212f5035b157.png)

**Comment by lwouis** [comment]
> Also, I've tried the arrow keys to handle data and it doesn't work for me, I clicked the circle of slider and press the arrow keys. can u please elaborate how to do it.

It used to be possible to navigate the preferences with keyboard (tab key) and select the sliders that way then arrow left/right, then tab to the next slider, etc. I tried right now and it didn't work in any tab of the preference window. Something is broken. I'll open a ticket to fix this.

Regarding your UI suggestion, I understand it's possible, but I know from having seen an implementation of such thing that's it's a rat nest of edge-cases, and back when I implemented the preferences, I thought a slider would avoid that. Just for fun here are some example use cases to cover. We need prevent the user from inputing in the box:

* Letters (people trying to type `one` instead of `1`, or `12px` instead of `12`)
* Special characters (including spaces, like typing ` 12` or `12 ` for example, or decimals like `1.5`)
* Emojis
* Non-English characters
* Handle numbers of other languages (I live in Japan where people would type `１２３` instead of `123`)
* Handle zero-leading notation like `01` instead of `1`
* Handling numbers too big and too small (user types `100000`. Should we block them from typing after the first `0`, or should we let them type it wrong but then have a red rectangle around the box?)
* The list goes on
